### ISSUE TITLE
fused_batch_norm's gradient implementation is incomplete

### ISSUE THREAD

**Issue by ppwwyyxx** [tensorflow/tensorflow#20407]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:N/A
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:N/A
- **TensorFlow installed from (source or binary)**:N/A
- **TensorFlow version (use command below)**:N/A
- **Python version**: N/A
- **Bazel version (if compiling from source)**:N/A
- **GCC/Compiler version (if compiling from source)**:N/A
- **CUDA/cuDNN version**:N/A
- **GPU model and memory**:N/A
- **Exact command to reproduce**:N/A

When `is_training=True`, the op looks like:
```python
y, batch_mean, batch_var = tf.nn.fused_batch_norm(x, scale, offset)
```
where `y`, `batch_mean` , and `batch_var` all depend on `x`. However the gradients of `batch_mean` and `batch_var` on `x` are currently None.

When `is_training=False`, the op looks like:
```python
y, _, _ = tf.nn.fused_batch_norm(x, scale, offset, mean, variance)
```
where `y` depend on all the inputs. However currently the gradients of `y` on `mean` and `variance` is None.

Both cases are not commonly used in practice, but mathematically there should be gradients.

**Comment by angersson** [comment]
@rmlarsen Could you take a look at this?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rmlarsen: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rmlarsen: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rmlarsen: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rmlarsen: It has been 59 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rmlarsen: It has been 74 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rmlarsen: It has been 89 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rmlarsen: It has been 104 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rmlarsen: It has been 119 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rmlarsen: It has been 134 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rmlarsen: It has been 149 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by luofan18** [comment]
Please implement gradient calculation for y depend on mean and variance for tf.nn.fused_batch_norm when is_training is set to False. I am implementing a cross-gpu batch normalization strategy for segmantic segmentation and hope to use this fused batch norm to perform batch normalization using mean and variance calculated across multiple gpu. However, fused_batch_norm do not return gradient for y depending on mean and variance if mean and variance are given. I opted out for the tf.nn.batch_normalization because it consume much more memory than fused_bathc_norm.

**Comment by ppwwyyxx** [comment]
Why does the status of the issue become "awaiting response"? It's a feature that's awaiting implementation.
